src/batou_ext/fcio.py

```diff
@@ -305,6 +309,69 @@ def get_diff(self, old, new):
         return result
 
 
+class DirectoryXMLRPC(batou.component.Component):
+    def configure(self):
+        self.xmlrpc = create_xmlrpc_client(self.environment)
+        self.provide("directory-xmlrpc", self.xmlrpc)
+
+
+class MaintenanceStart(batou.component.Component):
+    def configure(self):
+        self.require("needs-maintenance", strict=False, reverse=True)
+        self.xmlrpc = self.require_one("directory-xmlrpc")
+
+    def verify(self):
+        raise batou.UpdateNeeded()
+
+    def update(self):
+        change_maintenance_state(
+            self.xmlrpc, self.host.name, desired_state=True
+        )
+
+
+class MaintenanceEnd(batou.component.Component):
```

Comment:
This has the downside that setting it up is a little verbose: you have to add three additional components.
We may be able to cut it down to two by removing the `DirectoryXMLRPC` and setting up the client in each component.

Another idea I had was to use a decorator which injects `Start` end `End` into a component (and calls the component's `configure` in between), but this has the downside that the maintenance is left when the component itself is updated. I.e.

```python
@needs_maintenance
class Foobar(Component):
  def configure(self):
    # The RG is in maintenance when X & Y are deployed
    self += X()
    self += Y()
  def update(self):
    # when this is reached, we're not in maintenance anymore.
```

Thoughts?